src/components/textarea/textarea.tsx

```diff
@@ -0,0 +1,498 @@
+import { CSS, SLOTS, RESIZE_TIMEOUT } from "./resources";
+import {
+  connectLocalized,
+  disconnectLocalized,
+  LocalizedComponent,
+  NumberingSystem,
+  numberStringFormatter
+} from "../../utils/locale";
+import { createObserver } from "../../utils/observers";
+import {
+  componentLoaded,
+  LoadableComponent,
+  setComponentLoaded,
+  setUpLoadableComponent
+} from "../../utils/loadable";
+import {
+  connectMessages,
+  disconnectMessages,
+  setUpMessages,
+  T9nComponent,
+  updateMessages
+} from "../../utils/t9n";
+import { TextareaMessages } from "./assets/textarea/t9n";
+import { throttle } from "lodash-es";
+import { InteractiveComponent, updateHostInteraction } from "../../utils/interactive";
+
+/**
+ * @slot - A slot for adding text.
+ * @slot footer-start - A slot for adding a leading footer.
+ * @slot footer-end - A slot for adding a trailing footer.
+ */
+
+@Component({
+  tag: "calcite-textarea",
+  styleUrl: "textarea.scss",
+  shadow: true,
+  assetsDirs: ["assets"]
+})
+export class Textarea
+  implements
+    FormComponent,
+    LabelableComponent,
+    LocalizedComponent,
+    LoadableComponent,
+    T9nComponent,
+    InteractiveComponent
+{
+  //--------------------------------------------------------------------------
+  //
+  //  Element
+  //
+  //--------------------------------------------------------------------------
+  @Element() el: HTMLCalciteTextareaElement;
+
+  //--------------------------------------------------------------------------
+  //
+  //  Properties
+  //
+  //--------------------------------------------------------------------------
+
+  /**
+   * When `true`, the component is focused on page load.
+   *
+   * @mdn [autofocus](https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes/autofocus)
+   */
+  @Prop({ reflect: true }) autofocus = false;
+
+  /**
+   * When `true`, interaction is prevented and the component is displayed with lower opacity.
+   *
+   * @mdn [disabled](https://developer.mozilla.org/en-US/docs/Web/HTML/Attributes/disabled)
+   */
+  @Prop({ reflect: true }) disabled = false;
+
+  /**
+   * Specifies the placeholder text for the component.
+   *
+   * @mdn [placeholder](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/textarea#attr-placeholder)
+   */
+  @Prop() placeholder: string;
+
+  /**
+   * When `true`, the component's value can be read, but cannot be modified.
+   *
+   * @mdn [readOnly](https://developer.mozilla.org/en-US/docs/Web/HTML/Attributes/readonly)
+   */
+  @Prop({ reflect: true }) readonly = false;
+
+  /**
+   * Specifies the number or rows allowed.
+   *
+   * @mdn [rows](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/textarea#attr-rows)
+   */
+  @Prop({ reflect: true }) rows: number;
+
+  /**
+   * Specifies the number or columns allowed.
+   *
+   * @mdn [cols](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/textarea#attr-cols)
+   */
+  @Prop({ reflect: true }) cols: number;
```

Comment:
I've got a basic question/concern about the overall organization of this component. I do like that areas are split up using sectional comment headers, but would it be possible to alpha-sort where possible such as the Props and move lifecycles to the bottom of the component at the very least? Also, adding private to non-public properties might help differentiate what is public and what is not.

I know these are just my personal preferences so I'm curious as to what others think. I've seen similar organization as what I'm suggesting in other Calcite components (button, card and others seem to have started out this way but looks like they weren't kept up with the addition of messages props amongst other things), but I haven't found any Calcite guidelines for this specifically. I do know there is a [component boilerplate](https://github.com/Esri/calcite-components/blob/master/BOILERPLATE_COMPONENT.md), but it doesn't seem to be organized in any meaningful way itself and the [styleguide](https://github.com/ionic-team/stencil/blob/main/STYLE_GUIDE.md) also doesn't make mention of these things yet.